app/resources/game_content.py

```diff
@@ -36,3 +37,19 @@ def part_owner(device: str, user: str) -> bool:
         if service.running_port is not None and config["services"][service.name]["allow_remote_access"]:
             return True
     return False
+
+
+def dict2tuple(data: dict) -> Tuple[float, float, float, float, float]:
+    return data["cpu"], data["ram"], data["gpu"], data["disk"], data["network"]
+
+
+def calculate_speed(
+    edata: Tuple[float, float, float, float, float], rdata: Tuple[float, float, float, float, float]
+) -> float:
+    return (
```

Comment:
this function looks wrong. maybe return something like this instead:
```python
return sum(rdata) / sum(edata)
```
this should always be a value between 0 and 1 and can be used as the service speed

Reply:
done